### PR TITLE
Fix renderer crash from adblock page(removed stats from adblock webui)

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/13775

When navigting from adblock to other pages, renderer is crashed.
This crash comes from referring RFH to try update stats whlie
new naviagion occurs. 
As we don't need to show stats, stats updating logic is removed.

image

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Load any sites in advance
2. Load brave://adblock in the same tab
3. Navigate back to previous or type www.brave.com and check renderer is not crashed.

### CODE REVIEW

File: browser/ui/webui/basic_ui.cc [hunk 1]
```diff
@@ -50,65 +49,18 @@ content::WebUIDataSource* CreateBasicUIHTMLSource(
...
-  observer_.reset(
-      new BasicUIWebContentsObserver(this, web_ui->GetWebContents()));
   Profile* profile = Profile::FromWebUI(web_ui);
   content::WebUIDataSource* source = CreateBasicUIHTMLSource(profile, name,
       resource_map, resource_map_size, html_resource_id,
       disable_trusted_types_csp);
   content::WebUIDataSource::Add(profile, source);
 }
 
-BasicUI::~BasicUI() {
```
Comment (iefremov) [inline browser/ui/webui/basic_ui.cc:93]: it appears to me we don't need this class at all

File: browser/ui/webui/brave_adblock_ui.cc [hunk 2]
```diff
@@ -111,37 +113,7 @@ void AdblockDOMHandler::HandleUpdateCustomFilters(const base::ListValue* args) {
 BraveAdblockUI::BraveAdblockUI(content::WebUI* web_ui, const std::string& name)
     : BasicUI(web_ui, name, kBraveAdblockGenerated,
         kBraveAdblockGeneratedSize, IDR_BRAVE_ADBLOCK_HTML) {
-  Profile* profile = Profile::FromWebUI(web_ui);
-  PrefService* prefs = profile->GetPrefs();
-  pref_change_registrar_ = std::make_unique<PrefChangeRegistrar>();
```
Comment (iefremov) [inline browser/ui/webui/brave_adblock_ui.cc:116]: we don't need `#include "components/prefs/pref_change_registrar.h"`, pls check for other unused headers

File: browser/ui/webui/basic_ui.h [hunk 3]
```diff
@@ -42,18 +42,7 @@ class BasicUI : public content::WebUIController {
           bool disable_trusted_types_csp = false);
   ~BasicUI() override;
 
-  // Called when subclass can set its webui properties.
-  virtual void UpdateWebUIProperties() {}
-
- protected:
-  bool IsSafeToSetWebUIProperties() const;
-  content::RenderFrameHost* GetRenderFrameHost();
```
Comment (iefremov) [inline browser/ui/webui/basic_ui.h:50]: forward declaration for `content::RenderFrameHost` is not needed

**Review by iefremov — APPROVED** [review]

File: browser/ui/webui/basic_ui.cc (hunk 1 above)
Comment (simonhong) [reply browser/ui/webui/basic_ui.cc:93]: Ok, I'll do in a separate issue. https://github.com/brave/brave-browser/issues/13955

File: browser/ui/webui/basic_ui.h (hunk 3 above)
Comment (simonhong) [reply browser/ui/webui/basic_ui.h:50]: done.

File: browser/ui/webui/brave_adblock_ui.cc (hunk 2 above)
Comment (simonhong) [reply browser/ui/webui/brave_adblock_ui.cc:116]: done.

**Comment by simonhong** [comment]
Merged - only `AdBlockServiceTest.CosmeticFilteringProtect1p` on Windows is failed and it's known issue.

**Comment by bsclifton** [comment]
@simonhong does this need to be uplifted back to 1.20?

**Comment by simonhong** [comment]
@bsclifton I think so because stable also has this renerer crash. I'll push uplift PR.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.22.2 Chromium: 88.0.4324.146 (Official Build) nightly (64-bit)
-- | --
Revision | 406dc88511162d6598242f2c709be1414a042fb0-refs/branch-heads/4324@{#2088}
OS | Windows 10 OS Version 2009 (Build 19042.746)
```

* ensured that the crash doesn't occur when following the STR/Cases outlined via https://github.com/brave/brave-core/pull/7811#issue-567289705
* ensured that the crash doesn't occur when following the STR/Cases outlined via https://github.com/brave/brave-browser/issues/13775#issuecomment-767789784

**Pull request merged**